server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -118,6 +123,8 @@ public class SecretResource {
   @Consumes(APPLICATION_JSON)
   public Response createSecret(@Auth AutomationClient automationClient,
       @Valid CreateSecretRequestV2 request) {
+    permissionCheck.checkAllowedOrThrow(automationClient, Action.CREATE);
```

Comment:
It occurs to me that we could be more specific by including the type of thing being created, but we can deal with that later if/when we expand to other resource types.